cpp/CPP14Parser.g4

```diff
@@ -359,21 +364,27 @@ typeSpecifierSeq: typeSpecifier+ attributeSpecifierSeq?;
 trailingTypeSpecifierSeq:
 	trailingTypeSpecifier+ attributeSpecifierSeq?;
 
+simpleTypeLengthModifier:
+	Short
+	| Long;
+	
+simpleTypeSignednessModifier:
+	Unsigned
+	| Signed;
+
 simpleTypeSpecifier:
 	nestedNameSpecifier? theTypeName
 	| nestedNameSpecifier Template simpleTemplateId
-	| Char
-	| Char16
-	| Char32
-	| Wchar
+	| simpleTypeSignednessModifier
+	| simpleTypeSignednessModifier? simpleTypeLengthModifier+
+	| simpleTypeSignednessModifier? Char
+	| simpleTypeSignednessModifier? Char16
+	| simpleTypeSignednessModifier? Char32
+	| simpleTypeSignednessModifier? Wchar
 	| Bool
-	| Short
-	| Int
-	| Long
-	| Signed
-	| Unsigned
+	| simpleTypeSignednessModifier? simpleTypeLengthModifier* Int
```

Comment:
@Marti2203 This would also be related to your comment above....it would also match long long long long long int here....or even long short long short int.

Again, I'm not sure if the grammar should limit it or if should be done at a higher level. For example in the long long long int case, I could see it being valid on a 256 bit processor sometime in the future? (just like long long int on a 64 bit processor is valid).

Obviously long short long short is bogus for everyone...again...not sure where we should try to catch this.

Reply:
In my opinion, it should be caught, but I would like somebody else to give their opinion.

Reply:
I'm included to go with your suggestion @Marti2203. I'll work on change and if no one comments in the next day or so, I'll push it up.....

Reply:
I'm going to leave it as is for right now.  We can follow up with something better.  It at least recognizes these patterns, as it didn't do that before.